src/Spectre.libGaussianMixtureModelling/ExpectationRunnerRef.h

```diff
@@ -53,11 +55,16 @@ namespace Spectre::libGaussianMixtureModelling
         /// <param name="affilationMatrix">Matrix symbolising the probability of affilation
         /// of each sample to a certain gaussian component.</param>
         /// <param name="components">Gaussian components.</param>
-        ExpectationRunnerRef(DataType* mzArray, unsigned size, DataType**& affilationMatrix,
+        /// <exception cref="ArgumentNullException">Thrown when mzArray pointer are null</exception>
+        ExpectationRunnerRef(DataType* mzArray, unsigned size, Matrix& affilationMatrix,
             std::vector<GaussianComponent>& components)
             : m_pMzArray(mzArray), m_DataSize(size), m_Components(components)
             , m_AffilationMatrix(affilationMatrix)
         {
+            if (mzArray == nullptr)
+            {
+                throw ArgumentNullException("mzArray");
```

Comment:
We'll unify it in further iterations - I have exactly the same exception under another name on my branch :D